### PR TITLE
fix bof

### PR DESCRIPTION
**Pull request by BOB4Drone**
fix [#1668](https://github.com/mavlink/mavros/issues/1668)

### CODE REVIEW

File: mavros_extras/src/plugins/mag_calibration_status.cpp [hunk 1]
```diff
@@ -99,6 +99,10 @@ class MagCalStatusPlugin : public plugin::Plugin
     mavlink::common::msg::MAG_CAL_REPORT & mr,
     plugin::filter::SystemAndOk filter [[maybe_unused]])
   {
+    if (mr.comapss_id >= sizeof(calibration_show) {
```
Comment (vooon) [inline mavros_extras/src/plugins/mag_calibration_status.cpp:102]: Please fix misprint and use [`size() method`](https://en.cppreference.com/w/cpp/container/array/size).

**Review by vooon — CHANGES_REQUESTED** [review]
Overall it's good, just please fix build and test.

File: mavros_extras/src/plugins/mag_calibration_status.cpp (hunk 1 above)
Comment (BOB4Drone) [reply mavros_extras/src/plugins/mag_calibration_status.cpp]: I just fixed it
Can you check it again?

Comment (vooon) [reply mavros_extras/src/plugins/mag_calibration_status.cpp]: https://github.com/mavlink/mavros/runs/4485554446?check_suite_focus=true#step:4:548

Please fix CI.

File: mavros_extras/src/plugins/mag_calibration_status.cpp [hunk 2]
```diff
@@ -99,6 +99,10 @@ class MagCalStatusPlugin : public plugin::Plugin
     mavlink::common::msg::MAG_CAL_REPORT & mr,
     plugin::filter::SystemAndOk filter [[maybe_unused]])
   {
+    if (mr.compass_id >= calibration_show.size()) {
+      return ;
```
Comment (vooon) [inline mavros_extras/src/plugins/mag_calibration_status.cpp:103]: https://github.com/mavlink/mavros/runs/4486152998?check_suite_focus=true#step:4:1899

Please fix style.

**Review by vooon — CHANGES_REQUESTED** [review]

File: mavros_extras/src/plugins/mag_calibration_status.cpp (hunk 2 above)
Comment (BOB4Drone) [reply mavros_extras/src/plugins/mag_calibration_status.cpp:103]: What does it mean? Should I erase the gap between return and ;?

Comment (BOB4Drone) [reply mavros_extras/src/plugins/mag_calibration_status.cpp:103]: I couldn't build the environment for some reason, so I couldn't directly test the build. Sorry to bother you. :(

**Review by vooon — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**